source/develop/developer-guide/engine/engine-debug-non-obfuscated-ui.html.md

```diff
@@ -0,0 +1,40 @@
+---
+title: oVirt Engine Debug Non-Obfuscated UI
+authors: eraviv
+---
+
+# oVirt Engine Debug Non-Obfuscated UI
+
+## Debug Non-Obfuscated UI
+
+### Owner
+
+*   Name: Eitan Raviv (eraviv)
+*   Email: <eraviv@redhat.com>
+
+### Introduction
+
+Sometimes during development for the Web admin it is required to understand how different events for rendering, backend queries, garbage collection etc. influence the GUI behaviour and responsiveness.
+In these cases working with gwt-debug does not help because the debug session sends requests to the gwt code server at which time the UI in the browser freezes and waits for the response.
+During the wait the natural sequence of events is altered and it is not possible to get a real picture of what is happening in the browser.
+
+In these instances debugging the UI in the browser itself is the only viable option, but debugging the code in obfuscated form is next to impossible.
+
+## Debugging steps
+
+#### Building engine with non-obfuscated code
+
+- To get non-obfuscated javascript code in the browser, in ovirt-engine | frontend | webadmin | modules | pom.xml | gwt-maven-plugin | style: modify from 'OBF' to 'pretty'
```

Comment:
After https://gerrit.ovirt.org/c/102473/ is merged we can use the same trick as below DEV_EXTRA_BUILD_FLAGS_GWT_DEFAULTS="-Dgwt.style=pretty" also might be worth mentioning that there is another option called "detailed"